README.md

```diff
@@ -1,21 +1,97 @@
-### Assigning students to courses
+# Application - Assigning students to courses
 
-This program is a simple app that stores courses and students data and assigns students to a course.
+This program is a simple app that stores courses and students data and can assigns students to a course. It is a simple REST API written in python using Flask.
 
-IT is a simple REST API written in python using Flask.
+## REST API
 
-## What is REST API and Flask
+A REST (REpresentational State Transfer) API (Application Programming Interface) is a transfer of resources (data) from a Server to a Client over the internet. Through the REST API, the client can perform a GET, POST, PUT, DELETE request to the server.
 
-> what api provides to client
+A client uses an API to request for resources from the server. A server has the resources and transfers it to the client through an API. A resource can be any object for example, user, photo, number. An API allows one software to talk to another. It is a procedure allowing a client to access the data from a server. 
 
+An example of an API is "https://github.com/StephenDsouza90/demo-crud-sqlite-python-app" where:
+ 
+github.com/StephenDsouza90 - Endpoint URL
 
-##  The set up
+demo-crud-sqlite-python-app - Parameter
 
+## Flask
+
+Flask is a mirco web framework written in Python. 
+
+>>>>>> Need to write more on flask
+
+## Set up of rest.py 
+
+This app runs on a localhost (my computer) with an IP address of '0.0.0.0' (which maps to my computer) and a port 8080 as an endpoint.
+
+Example: 
+
+github.com - Localhost
+
+StephenDsouza90 - Endpoint
```

Comment:
This is wrong. Endpoint means IPAddress with a Port combination. Or DNS name (like google.com)  with a port. By default the port is 80 and it is never shown.

So google.com:80 would point to google.com and this is the endpoint.

The endpoint doesn't have and Paths or resourses. It's always either a DNS name or and IPAddress:Port combination like localhost:8080 is the endpoint.

Endpoint means a way to contact a server. A server can be contacted via the IP address or DNS Name and it's port usually by default 80. Where the IP Address or DNS name locates the server over the internet, it's like a server's ID and the port is thing on the server's machine where the server can listen to incoming requests from the client and process it.

Reply:
change made to endpoint